### ISSUE TITLE
Scan Bluetooth (non BLE) devices

### ISSUE THREAD

**Issue by abmantis** [esphome/feature-requests#65]
**Describe the problem you have/What new integration you would like**

It would be great to be able to scan classic bluetooth (not BLE) devices, like phones.

**Please describe your use case for this integration and alternatives you've tried:**

A device tracker for phones.

**Additional context**

I'm not sure if there's a high-level API for this yet, but it is possible: https://techtutorialsx.com/2018/03/04/esp32-arduino-bluetooth-classic-getting-started/

The PR mentioned on that article is already merged, so it should be simpler.

**Comment by abmantis** [comment]
Quick question: are you guys using (or OK with using) the espidf framework from platformio? I'm working on a classic BT device tracker for the ESP32, which could probably be included in esphome, but it requires that framework.

**Comment by OttoWinter** [comment]
ESPHome uses the arduino framework, but that uses esp-idf internally, so yes that can be used (see ble_tracker implementation for example)

**Comment by abmantis** [comment]
@OttoWinter cool, thanks! I'll try to do something then!

**Comment by runevad** [comment]
If this would be implemented, does it mean we can use an ESP32 and Esphome to set up Room tracking like: https://www.home-assistant.io/components/sensor.mqtt_room/  ? 
That would be awesome :)

**Comment by abmantis** [comment]
@runevad yes, that was my plan. Unfortunately I'm finding it hard to do without having an impact on the phone's baterry, since the classic bluetooth discovery APIs from espressif are incomplete at the moment.

**Comment by GeoffAtHome** [comment]
Would be great if you could detect regular Bluetooth devices as easily as Bluetooth LE device.

**Comment by nmsoares** [comment]
Any developments on this ?

**Comment by deex2** [comment]
Hi,

Any developments on this please?

**Comment by martydingo** [comment]
Any progress? 

It's been years. And I have a device that requires pairing before BLE can work. I am seeing a status=133 error code, and I can't seem to get the ESP32 to pair with the BLE device.
